### PR TITLE
Added the ability to add Resource Attributes for all the scrappers

### PR DESCRIPTION
**Pull request by ishleenk17**
With this change, resource attributes can be added to the outgoing metrics. 
they will be added as part of the attributes of the datapoint. 

If they are added as a generic Resource Attribute, it gets added to the OTEL native document as well, which is not expected.

### CODE REVIEW

**Comment by ishleenk17** [comment]
@tommyers-elastic : I see a failure here. Looks like its looking for a Makefile. 
I see Makefiles are present in other Processors. Do we need to add one here as well?
But another question, how was it working/compiling before.

File: processor/elasticprocessor/internal/hostmetrics/cpu.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"go.opentelemetry.io/collector/pdata/pmetric"
 )
 
-func addCPUMetrics(metrics pmetric.MetricSlice, dataset string) error {
+func addCPUMetrics(metrics pmetric.MetricSlice, rm pcommon.Resource, dataset string) error {
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/cpu.go]: nit: can we just call this `resource`, so it's more obvious what it is when it's used later?

File: processor/elasticprocessor/internal/hostmetrics/metric.go [hunk 2]
```diff
@@ -47,6 +47,23 @@ func addMetrics(ms pmetric.MetricSlice, dataset string, metrics ...metric) {
 			dp.SetStartTimestamp(metric.startTimestamp)
 		}
 
+		process_ppid, _ := rm.Attributes().Get("process.parent_pid")
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/metric.go]: shouldn't these be added in the process specific code?

File: processor/elasticprocessor/internal/hostmetrics/process.go [hunk 3]
```diff
@@ -129,8 +129,10 @@ func addProcessMetrics(metrics pmetric.MetricSlice, dataset string) error {
 	cpuTimeValue = total * 1000
 	systemCpuTime = systemCpuTime * 1000
 	userCpuTime = userCpuTime * 1000
+	timeDiff = timestamp.AsTime().UnixMilli() - startTime
```
Comment (tommyers-elastic) [inline processor/elasticprocessor/internal/hostmetrics/process.go]: i was confused for a second about what this value is - can we give it a more descriptive name?

perhaps `processRuntime` or something?

File: processor/elasticprocessor/internal/hostmetrics/metric.go (hunk 2 above)
Comment (tommyers-elastic) [reply processor/elasticprocessor/internal/hostmetrics/metric.go]: additionally, what's the reason for copying these resource attributes to data point attributes?

Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/metric.go]: If you remember we discussed, that if we make it part of resource attributes, it gets added to both OTEL and ES specific documents. For it to be present only for ES documents, we decided to add it to DP attributes.

Its now added to the process specific code.

File: processor/elasticprocessor/internal/hostmetrics/cpu.go (hunk 1 above)
Comment (ishleenk17) [reply processor/elasticprocessor/internal/hostmetrics/cpu.go]: Done

File: processor/elasticprocessor/internal/hostmetrics/metric.go (hunk 2 above)
Comment (tommyers-elastic) [reply processor/elasticprocessor/internal/hostmetrics/metric.go]: ah right, thanks for the reminder.

**Review by tommyers-elastic — APPROVED** [review]

**Pull request merged**